<!-- EVALUATE VISUAL ODOMETRY -->
<launch>

    <!-- #################
            BAG PLAYER
         ################# 
     MUST HAVE
     /stereo_cam/left/image_raw
     /stereo_cam/left/camera_info
     /stereo_cam/right/image_raw
     /stereo_cam/right/camera_info
     -->
	
    <arg name="calibrated" default="1" />
    <arg name="respawn" default="true" />
    <arg name="decimation_x" default="1"/>
    <arg name="decimation_y" default="1"/>  
    <arg name="camera_nodelet_manager_name" default="cam_manager" />
    <param name="use_sim_time" value="true" />
    
    <!-- #############################################
            GET VISUAL ODOMETRY AND GROUND TRUTH DATA
         ############################################# -->
    <node pkg="evaluate_vo" type="get_vogt_node" name="get_stereo" args="/home/anaritapereira/evaluate_vo_plots/set27/bag2 stereo odometry gps" output="screen">
        <param name="topicname_vo" value="viso_odometer/odometry" />
        <param name="topicname_gps" value="asterx2_node/lrm_gps" />
        <param name="topicname_imu" value="imu/data" />
    </node>

    <param name="use_sim_time" value="true"/>
    
    <!-- ##############################
            SPLIT AND PROCESS IMAGES
         ############################## -->
         
    <node pkg="split_and_process" type="split_and_process_node" name="split_and_process" output="screen">
        <remap from="image" to="/StereoCam/TriImage" />
        <param name="input0/camera_info_url" value="package://evaluate_vo/params/cameras/calib/00b09d0100bec852_wide_left_8_6.yaml" />
	    <param name="input1/camera_info_url" value="package://evaluate_vo/params/cameras/calib/00b09d0100bec852_narrow_right.yaml" />
        <param name="input2/camera_info_url" value="package://evaluate_vo/params/cameras/calib/00b09d0100bec852_wide_right_8_6.yaml" />
    </node>
    
    
    <!-- ###################
            Transformations
         ################### -->

    <node pkg="tf2_ros" type="static_transform_publisher" name="imu_link" args="-1.31 0 1.20 0 0 0 1 base_link imu" />
    
    <node pkg="tf2_ros" type="static_transform_publisher" name="cam_link" args="-0.30 0 1.49 -1.57 0 -1.67  base_link camera" />
    
    <node pkg="tf2_ros" type="static_transform_publisher" name="vodom_link" args="0 0 0 0 0 0 1  base_link vodom" />

    
    <!-- #################
            Viso2
         ################# -->
    <node pkg="viso2_ros" type="stereo_odometer" name="viso_odometer" output="screen" respawn="$(arg respawn)" >
        <remap from="stereo" to="StereoCam/wide"/>
        <remap from="image" to="image_rect"/>
        <param name="base_link_frame_id" value="base_link"/>
        <param name="odom_frame_id" value="vodom"/>
        <param name="sensor_frame_id" value="camera"/>
		<param name="bucket_width" value="1280"/>
		<param name="bucket_height" value="960"/>
		<param name="max_features" value="1000"/>
    </node> 

</launch>
